tests-web/www/js/main.js

```diff
@@ -1,7 +1,230 @@
+    try {
+      const resp = await fetch(url);
+      const osmInput = await resp.text();
+
+      importButton.innerText = "Importing OSM data...";
+
+      // TODO Ask overpass
+      const drivingSide = "Right";
+
+      const network = new JsStreetNetwork(osmInput, {
+        driving_side: drivingSide,
+      });
+      const rawMapLayer = makePlainGeoJsonLayer(network.toGeojsonPlain());
+      const bounds = rawMapLayer.getBounds();
+
+      var layers = [];
+      layers.push(app.addLayer("Geometry", rawMapLayer));
+      layers.push(
+        app.addLayer(
+          "Detailed geometry",
+          makeDetailedGeoJsonLayer(network.toGeojsonDetailed())
+        )
+      );
+      layers.push(app.addLayer("OSM", makeOsmLayer(osmInput)));
+      // TODO The dot layer
+
+      return new TestCase(app, null, osmInput, drivingSide, layers, bounds);
+    } catch (err) {
+      window.alert(`Import failed: ${err}`);
+    } finally {
+      // Make the button clickable again
+      importButton.innerText = "Import current view";
+      importButton.disabled = false;
+    }
+  }
+
+  renderControls(container) {
+    container.innerHTML = "";
+    if (this.name) {
+      const title = container.appendChild(document.createElement("b"));
+      title.innerText = `Currently showing ${this.name}`;
+
+      const button = container.appendChild(document.createElement("button"));
+      button.type = "button";
+      button.innerHTML = "Reimport";
+      button.onclick = () => {
+        button.disabled = true;
+        this.reimport();
+      };
+    } else {
+      container.innerHTML = `<b>Custom imported view</b>`;
+    }
+
+    const button = container.appendChild(document.createElement("button"));
+    button.type = "button";
+    button.innerHTML = "Download osm.xml";
+    button.onclick = () =>
+      downloadGeneratedFile(`${this.name || "new"}.osm.xml`, this.osmXML);
+  }
+
+  reimport() {
+    try {
+      const network = new JsStreetNetwork(this.osmXML, {
+        driving_side: this.drivingSide,
+      });
+      this.layers.push(
+        this.app.addLayer(
+          "Geometry (reimport)",
+          makePlainGeoJsonLayer(network.toGeojsonPlain())
+        )
+      );
+      this.layers.push(
+        this.app.addLayer(
+          "Detailed geometry (reimport)",
+          makeDetailedGeoJsonLayer(network.toGeojsonDetailed())
+        )
+      );
+      // TODO The dot layer
+    } catch (err) {
+      window.alert(`Reimport failed: ${err}`);
+    }
+  }
+}
+
+function setupLeafletMap(mapContainer) {
+  const map = L.map(mapContainer, { maxZoom: 21 }).setView([40.0, 10.0], 4);
+  L.tileLayer("https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png", {
+    maxNativeZoom: 18,
+    maxZoom: 21,
+    attribution: "© OpenStreetMap",
+  }).addTo(map);
+  // Geocoder, satellite layers, etc
+  return map;
+}
+
+// TODO Port stuff
 const useMap = (map) => {
   const container = map.getContainer();
   container.ondrop = makeDropHandler(map);
```

Comment:
I haven't fixed this up yet. I don't think I've used this before -- you could drag and drop a `.osm` or `.json` or `.dot` file into the window and have it add a layer, right? How should that work with the `currentTest` paradigm -- maybe we try to create a new test case? But we won't always have the OSM XML as input, and we need that for every `TestCase`. So maybe this just blindly adds a layer to the current test?